src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeFileBasedSource.scala

```diff
@@ -216,6 +220,127 @@ class DeltaLakeFileBasedSource(private val spark: SparkSession) extends FileBase
         None
     }
   }
+
+  private def deltaLakeVersionHistory(index: IndexLogEntry): Seq[(Int, Int)] = {
+    val versions =
+      index.derivedDataset.properties.properties.getOrElse(DELTA_VERSION_HISTORY_PROPERTY, "")
+
+    versions.split(",").reverseIterator.foldLeft(Seq[(Int, Int)]()) { (list, versionStr) =>
+      val pair = versionStr.split(":")
+      val versions = (pair(0).toInt, pair(1).toInt) // (index log version, delta table version)
```

Comment:
This comment could be part of the code using pattern matching on assignment, e.g.

```
val Array(indexLogVersion, deltaTableVersion) = versionStr.split(":").map(_.toInt)
```